iree/compiler/Dialect/HAL/IR/HALOps.cpp

```diff
@@ -1500,6 +1500,24 @@ ArrayAttr InterfaceOp::getExecutableSetLayoutsAttr() {
       })));
 }
 
+bool InterfaceOp::isEquivalentTo(InterfaceOp other) {
+  auto bindings = getBlock().getOps<InterfaceBindingOp>();
+  auto otherBindings = other.getBlock().getOps<InterfaceBindingOp>();
+  auto it = bindings.begin(), lhsEnd = bindings.end();
```

Comment:
I'm sure there's some ADT helper for this; like
```
auto bindings = llvm::to_vector<4>(...);
auto otherBindings = llvm::to_vector<4>(...);
return bindings.size() == otherBindings.size() &&
       llvm::all_of(llvm::zip(bindings, otherBindings), [](InterfaceBindingOp lhs, InterfaceBindingOp rhs) {
         return OperationEquivalence::isEquvalentTo(lhs, rhs);
       });
```
(@silvasean may know a cleaner way)

Reply:
I was probably looking at https://github.com/llvm/llvm-project/blob/a67d164a82bc535025bb65736ee4e5f381500ed8/mlir/lib/Transforms/Utils/RegionUtils.cpp#L484-L497 when I wrote this originally.

Reply:
Ok, found similar code here using the structure you suggested: https://github.com/llvm/llvm-project/blob/f402e682d0ef5598eeffc9a21a691b03e602ff58/mlir/lib/Dialect/Traits.cpp#L21-L24 and updated to it.